### PR TITLE
Store taxa import copy & add vector and continent capabilities to region definition

### PR DESCRIPTION
**Pull request by RRTogunov**
# Why have changes been made?

- If rerunning code from previous date-accessed (eg if troubleshooting code), it would be helpful to use the same taxa as used previously.
- add functionality to load load multiple regions from csmaps package, and download countries or continents using rnaturalearth package.

# What changes have been made?

- pipeline/import/taxaImport.R - create dateAccessed folder up front, check if taxaImport exists in dateAccessed folder, otherwise, import from data/external and save copy
- functions/defineRegion.R - allow for selecting multiple regions within Norway using %in%, import countries/continents using rnaturalearth package, & join multiple shapes into single boundary.
- masterScript.R - add comment specifying that vector of locations can be provided

### CODE REVIEW

File: functions/defineRegion.R [hunk 1]
```diff
@@ -47,5 +52,8 @@ defineRegion <- function(level = "county", region = "50", runBuffer = FALSE, ext
   # Align project coordinates with the rest of our polygons.
   regionGeometry <- st_transform(regionGeometry, crs =  "+proj=longlat +ellps=WGS84")
   
+  # combine multipolygon into single sf polygon
+  regionGeometry <- st_union(regionGeometry)
```
[the remark this answers is not in the document]
Comment (samaperrin) [reply functions/defineRegion.R:56]: Europe also looks a bit strange (screenshot attached)
image

**Review by samaperrin — CHANGES_REQUESTED** [review]
Some issues on the continent scale for Europe and Africa

File: functions/defineRegion.R [hunk 2]
```diff
@@ -14,18 +14,23 @@
...
+    library(csmaps)
     regionCode <- paste0("county_nor", region)
-    regionGeometry <- nor_county_map_b2020_default_sf$geometry[nor_county_map_b2020_default_sf$location_code == regionCode]
+    regionGeometry <- nor_county_map_b2020_default_sf$geometry[nor_county_map_b2020_default_sf$location_code %in% regionCode]
   } else if (level == "country") {
-    world <- giscoR::gisco_get_countries()
-    country <- world[world$NAME_ENGL == region,]
-    regionGeometry <- st_transform(country$geometry,crs = "+proj=longlat +datum=WGS84 +no_defs")
+    library(rnaturalearth)
+    regionGeometry <- regionGeometry <- ne_countries("large", type = "map_units", geounit = region, returnclass = "sf")
```
[the remark this answers is not in the document]
Comment (RRTogunov) [reply functions/defineRegion.R:28]: nope. deleting.

File: pipeline/import/taxaImport.R [hunk 3]
```diff
@@ -42,6 +30,24 @@ if (!file.exists(folderName)) {
...
+if (!exists("region")) {region <- "Norway"}
+regionGeometry <- defineRegion(level, region)
+
+# Define initial species list.
+if(file.exists(paste0(folderName, "/focalTaxa.csv"))){
+  focalTaxon <- read.csv(paste0(folderName, "/focalTaxa.csv"), header = T)
+} else {
+  focalTaxon <- read.csv("data/external/focalTaxa.csv", header = T)
+  focalTaxon <- focalTaxon[focalTaxon$include,]
+  focalTaxa <- focalTaxon$taxa
```
[the remark this answers is not in the document]
Comment (RRTogunov) [reply pipeline/import/taxaImport.R:46]: Yes, I think L46 can be deleted. also, for better reproducibility, I think I should move L45 after copying external/focalTaxa.csv to <folderName>/focalTaxa.csv.

File: functions/defineRegion.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (RRTogunov) [reply functions/defineRegion.R:56]: fixed invalid geometries.

**Review by samaperrin — APPROVED** [review]
All works a treat now, nice work!

**Pull request merged**